bitnami/thanos/templates/storegateway/statefulset-sharded.yaml

```
@@ -0,0 +1,257 @@
+{{- if and .Values.storegateway.enabled .Values.storegateway.sharded.enabled }}
+
+{{- $shards := int 0 }}
+{{- $hashPartioning := false }}
+{{- $timePartioning := false }}
+
+{{- if .Values.storegateway.sharded.hashPartioning.shards }}
+  {{- $shards = int .Values.storegateway.sharded.hashPartioning.shards }}
+  {{- $hashPartioning = true }}
+{{- else }}
+  {{- $shards = len .Values.storegateway.sharded.timePartioning }}
+  {{- $timePartioning = true }}
+{{- end }}
+
+{{- range $index, $_ := until $shards }}
+apiVersion: {{ include "common.capabilities.statefulset.apiVersion" $ }}
+kind: StatefulSet
+metadata:
+  name: {{ include "common.names.fullname" $ }}-storegateway-{{ $index }}
+  labels: {{- include "common.labels.standard" $ | nindent 4 }}
+    app.kubernetes.io/component: storegateway
+    {{- if $.Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+spec:
+  replicas: {{ $.Values.storegateway.replicaCount }}
+  podManagementPolicy: {{ $.Values.storegateway.podManagementPolicy }}
+  serviceName: {{ include "common.names.fullname" $ }}-storegateway-headless
```

Same here. Truncate:
```suggestion
  serviceName: {{ printf "%s-storegateway-headless-%s" (include "common.names.fullname" $) $index | trunc 63 | trimSuffix "-" }}
```

---

would this not break, the service does not have an index

---

Then we need to figure out how to trim the name while keeping the index. For instance:
```
  {{- $svcNamePrefix :=  printf "%s-storegateway-headless"  (include "common.names.fullname" $) | trunc 61 | trimSuffix "-" }}
  serviceName: {{ printf "%s-%s" $svcNamePrefix }}
```

---

Please review this